### PR TITLE
Add name to the kernelOptions

### LINKED ISSUE #43: R kernel is not loaded by `thebe` in documentation
It's probably a minor configuration issue that is causing this to happen on the [configure page](https://sphinx-thebe.readthedocs.io/en/latest/configure.html): 

The kernel is set to `ir`, so maybe it should be just "r"? 

`thebe-kernel: ir`

Screen Shot 2022-01-31 at 2 07 12 AM

Thanks!

### LINKED ISSUE #53: Your sphinx documentation is not working
I am trying to run r-kernel thebe. So I came up to your site. However, your own documentation is giving me the same error I am currently facing:

```
NameError: name 'library' is not defined
```

Although you have specified an R kernel, it is running on Jupyter... Any ideas or workarounds would be highly appreciated

### LINKED ISSUE #59: Change config option name to correctly select kernel
### Describe the bug

**context**

I have a jupyter-book using octave notebooks. Live-Code with thebe stopped working after updating jb to 0.13.1.

**expectation**.

Running live-code with thebe should work out of the box with an octave kernel, but it doesn't. Running the jupyter notebooks in mybinder.org works as expected.

**bug**

This seems to be caused by https://github.com/executablebooks/thebe/issues/349. If I understand correctly, thebe >= 0.6.0 requires the kernel to be set via `kernelOptions.name` rather than `kernelOptions.kernalName`. I verified this by manually adding `kernelOptions.name` to a locally built html file.

**possible fix**

After a first glance at sphinx-thebe, I suppose it could suffice to fix this line:

https://github.com/executablebooks/sphinx-thebe/blob/d9c1f3db9a65146319c15bdc7d4e91d25c58f23e/sphinx_thebe/__init__.py#L148

### Reproduce the bug

As a consequence, the `sphinx-thebe` documentation is currently broken: Running the interactive R code on https://sphinx-thebe.readthedocs.io/en/latest/configure.html does not work.

### List your environment

jupyter-book version 0.13.1

### PR DESCRIPTION
**Pull request by joergbrech**
Fixes #59.

See also https://github.com/executablebooks/thebe/issues/349.

Follow-up question: Can we have some kind of doctests verifying that all code blocks in the documentation don't error?

Edit from @choldgraf : i think this also closes #43 and closes #53

Edit: Closes https://github.com/executablebooks/jupyter-book/issues/1173

### CODE REVIEW

**Review by choldgraf — APPROVED** [review]
Good catch - do you think we can deprecate kernelName or should we keep it?

For tests, i agree, but nobody has had the bandwidth to write them. I guess you could do it with playwright or something like this.

**Comment by joergbrech** [comment]
> Good catch - do you think we can deprecate kernelName or should we keep it?

I honestly don't know :shrug: I wasn't brave enough to deprecate it.

**Comment by choldgraf** [comment]
@stevejpurves would you be willing to provide guidance?

**Comment by stevejpurves** [comment]
I would keep it -- I believe that a kernel launch can fail if the (default) named kernel is not available on the server/service, and if for some reason a kernel named `python3` is not there then that would block usage, even though the public binder may have that kernel available.

**Comment by choldgraf** [comment]
@stevejpurves so `name` and `kernelName` serve two different purposes then? Could you link to a doc (or provide briefly here) so that we can cross-link in the sphinx-thebe docs to help others understand when/how to use?

**Comment by choldgraf** [comment]
ah looking at the comment in the linked issue, it seems that `name` is the new correct option after thebe 0.6, and `kernelName` is an old / deprecated option:

- https://github.com/executablebooks/thebe/issues/349

Does that sound correct to you @stevejpurves ?

**Comment by joergbrech** [comment]
I am not entirely sure, but I believe this could fix https://github.com/executablebooks/jupyter-book/issues/1173 also.

**Comment by choldgraf** [comment]
Yep i think you are right. Let's cross link that issue in the PR as well so that it closes it. We can make a release quickly!

**Comment by stevejpurves** [comment]
@choldgraf I can't actually find any reference to the two existing side by side outside of `thebe`. 
As`kernelName` has no effect in the current thebe release, and as thebe is moving to not using the Juptyerlab sessions at all - it looks like deprecating `kernelName` would be ok here after all, and as a breaking change make it clearer for `sphinx-thebe` users to switch to /use`name`.

In the meantime, thebe `0.8.x` needs an update to provide the backwards compatbility

**Comment by choldgraf** [comment]
Ah ok, thanks for this clarification - I'd then suggest:

- make kernelName rename itself to name if given, and raise a warning.
- Add a comment to that line that the 0.8 release of thebe will change its API. @stevejpurves it would be really helpful if you took a look at this extension just to recommend if there are any other things that will be deprecated)
- open an issue to track the future change

**Comment by joergbrech** [comment]
> make kernelName rename itself to name if given, and raise a warning.

@choldgraf I am afraid I don't understand enough about the interplay of thebe/sphinx/jupyter etc. What do you mean by this? Doesn't sphinx-thebe just parse `thebe-kernel` from the Markdown metadata and copy it to the thebe html config? Where would `kernelName` be given, so that it can be renamed?

**Comment by choldgraf** [comment]
@joergbrech ah I'm being silly - `kernelName` is chosen by `sphinx-thebe`, not the user. I think we can just deprecate the use of `kernelName` entirely by deleting that line.

File: sphinx_thebe/__init__.py [hunk 1]
```diff
@@ -146,6 +146,7 @@ def update_thebe_context(app, doctree, docname):
         }},
         kernelOptions: {{
             kernelName: "{kernel_name}",
```
Comment (choldgraf) [inline sphinx_thebe/__init__.py:148]:
```suggestion
```

File: tests/test_build/config_custom.html [hunk 2]
```diff
@@ -11,6 +11,7 @@
         },
         kernelOptions: {
             kernelName: "ir",
```
Comment (choldgraf) [inline tests/test_build/config_custom.html:13]:
```suggestion
```

File: tests/test_build/config_index.html [hunk 3]
```diff
@@ -11,6 +11,7 @@
         },
         kernelOptions: {
             kernelName: "python3",
```
Comment (choldgraf) [inline tests/test_build/config_index.html:13]:
```suggestion
```

**Comment by stevejpurves** [comment]
@choldgraf yes I will - how about you go ahead and release this change first, then I can sit down and test the jb|sphinx-thebe|thebe together and make sure I have a clear picture.

**Comment by joergbrech** [comment]
@choldgraf just a friendly ping. Are we ready to merge?

**Review by choldgraf — APPROVED** [review]
Looks good, thanks!

**Comment by choldgraf** [comment]
Yep - thanks for the ping. If any of you are interested in having merge rights on this repository, I am happy to provide them. I clearly do not have enough time to maintain this in addition to all the other stuff I have to do, and I don't want to block progress on myself!

**Comment by joergbrech** [comment]
> Yep - thanks for the ping. If any of you are interested in having merge rights on this repository, I am happy to provide them. I clearly do not have enough time to maintain this in addition to all the other stuff I have to do, and I don't want to block progress on myself!

Thanks, but as this is just a small side project, I don't see myself contributing often. Also I know too little about the whole jupyter-sphinx-thebe ecosystem to be of much value. Once this is in jupyter book I will be happy for a while 🙂

**Pull request merged**